dspMetadataGUI/util/metaDataSet.py

```diff
@@ -0,0 +1,1290 @@
+    """
+    Grant Shape.
+
+    Corresponds to `dsp-repo:Grant` in the ontology.
+    """
+
+    def __init__(self, meta):
+        self.meta = meta
+        self.iri_suffix = IRIFactory.get_unique_iri('grant', meta)
+
+        self.name = Property(meta, "Name",
+                             "Name of the grant",
+                             "Ambizione",
+                             Datatype.STRING,
+                             Cardinality.ZERO_OR_ONE,
+                             predicate=dsp_repo.hasName)
+
+        self.url = Property(meta, "URL",
+                            "URL of the grant",
+                            "https://www.snf.ch/grants/001",
+                            Datatype.URL,
+                            Cardinality.ZERO_OR_ONE,
+                            predicate=dsp_repo.hasURL)
+
+        self.number = Property(meta, "Number",
+                               "The number of the grant.",
+                               "00012345",
+                               Datatype.STRING,
+                               Cardinality.ZERO_OR_ONE,
+                               predicate=dsp_repo.hasNumber)
+
+        self.funder = Property(meta, "Funder",
+                               "Funding person or institution of the project",
+                               "",
+                               Datatype.PERSON_OR_ORGANIZATION,
+                               Cardinality.ONE_TO_UNBOUND,
+                               predicate=dsp_repo.hasFunder)
+
+    def get_properties(self):
+        # LATER: if Property is in another file and imported, this can be typed
+        """
+        Get all properties held by the object.
+
+        Returns:
+            List[Property]: A list with all the Properties held by the object.
+        """
+        return [
+            self.funder,
+            self.name,
+            self.number,
+            self.url,
+        ]
+
+    def __str__(self):
+        classname = str(self.get_rdf_iri()).split('#')[1]
+        n1 = "<funder missing>"
+        v = self.funder.value
+        if v:
+            if isinstance(v, list):
+                v = v[0]
+            n1 = str(v)
+        return f"{classname}: [{n1}]"
+
+
+class Property():
+    """
+    General representation of a property.
+
+    Corresponds to `sh:property`
+    """
+
+    def __init__(self, meta: MetaDataSet, name: str, description: str, example: str, datatype: Datatype.STRING,
+                 cardinality=Cardinality.UNBOUND, value=None, value_options=None,
+                 predicate=dsp_repo.whatever, multiline=False):
+        self.meta = meta
+        self.name = name
+        self.description = description
+        self.example = example
+        self.datatype = datatype
+        self.cardinality = cardinality
+        self.value = value
+        self.value_options = value_options
+        self.predicate = predicate
+        self.multiline = multiline
+
+    def get_url_property_id(url: str) -> str:
```

Comment:
It is aloud to leave out `self` parameter in a method?

Reply:
it shouldn't cause any issues, as `self` never gets called here (so it could be a static method too), but I added it in anyways, just to be on the safe side.

In any case, this method should get removed soon, so the problem will solve itself in that sense.